project-loader/src/main/kotlin/Loader.kt

```diff
@@ -100,7 +100,7 @@ fun <T> executeWithProject(project: File,
 
     val cfg = basicEnvironmentConfig()
 
-    plugins.forEach { cfg.addPreInstalledPlugin(it.path, it.id) }
+    plugins.forEach { cfg.addPlugin(it.path, it.id) }
```

Comment:
@coolya Sorry but tbh. I can't follow entirely. The only thing this PR is changing is how the plugins configured in the `plugins` property are loaded. 
Before this commit the project loader was expected them to be placed under `$MPS_HOME/plugins`. 
Jugding from the past, this was not required for the mps-gradle-plugin in version  `1.2.177.46341c0`.

However if the privious implementation was correct, we should then at least update the documentation so that users know where to place their custom plugin. And this location would then always be `$MPS_HOME/plugins`.

Reply:
I know that it is changing it, but there was a reason why this change was introduced by Wladi, he was unable to load pre installed plugins like the http support. This was most probably required because of a change in platform where many things moved from relative to absolute paths I don't recall all the details but there was a PR where this is discussed. 

So the question I want to get answered is: 

- is the path property in the plugin always relative to the `pluginLocation`? 
- is the path property in the plugin a absolute path? 

if either of the above is true we need to have a way to load pre installed plugins via the `plugins` list.

Reply:
> is the path property in the plugin a absolute path?
If I check the MPS sources for 2019.3 I can see that `env.addPlugin()` is using an absolut path for the path property. see: https://github.com/JetBrains/MPS/blob/2019.3/core/tool/environment/source_gen/jetbrains/mps/tool/environment/EnvironmentConfig.java#L68

Reply:
Ok I see, so `pluginLocation` is unused now. Then this PR needs changes to address the fact that plugins are always referenced with their absolute path. 

IMHO we need a way to declare a plugin is "pre installed" in the `plugins` list depending on that the logic need to call `addPreInstalledPlugin` for pre installed ones and `.addPlugin` for not pre installed plugins. In order construct the correct path for plugin in it needs to use `pluginLocation` as a prefix to the `path` defined in the `Plugin` of the `plugins` list otherwise we would need to make all plugin paths absolute which is infeasible because the caller of the plugin might now have control about the location where MPS is extracted.

Reply:
Agree